src/_pytest/mark/structures.py

```diff
@@ -344,25 +345,22 @@ def with_args(self, *args: object, **kwargs: object) -> "MarkDecorator":
         mark = Mark(self.name, args, kwargs, _ispytest=True)
         return MarkDecorator(self.mark.combined_with(mark), _ispytest=True)
 
-    # Type ignored because the overloads overlap with an incompatible
-    # return type. Not much we can do about that. Thankfully mypy picks
-    # the first match so it works out even if we break the rules.
     @overload
-    def __call__(self, arg: Markable) -> Markable:  # type: ignore[misc]
+    def __call__(self, arg: Markable) -> Markable:
         pass
 
     @overload
-    def __call__(self, *args: object, **kwargs: object) -> "MarkDecorator":
+    def __call__(self, *args: Any, **kwargs: Any) -> "MarkDecorator":
         pass
 
-    def __call__(self, *args: object, **kwargs: object):
+    def __call__(self, *args: Any, **kwargs: Any) -> Union[Markable, "MarkDecorator"]:
```

Comment:
`Markable` is an unbound `TypeVar` here, which usually doesn't make sense. Since the typing in covered by the overloads I chose to omit the return type here.

Reply:
Is it unbound? I figured mypy would use the overload, but I can adjust this so there is less overlap. I can force non keyword arguments but without knowing at least one required keyword argument I don't know how to tell mypy that at least one has to be provided. to force the `MarkDecorator` path.

Reply:
Just to follow up on the last part of your comment... As you have it mypy (possibly any type checker) will read this as `-> Any`. I had thought as you are on a different codebase and I noticed that mypy wasn't checking the function at all (I had left off the argument signature too), I then added the argument signature and tested that I could return anything even outside the bounds of the `@overload`s. This might have been fixed in mypy, but it's best to be explicit about your return type. When I had added the return annotation it required casting the return in the `Markable` path since it was now checking the return was not `Any`.

Reply:
When a function has overloads defined, the typing of the implementation function is only used for the implementation -- it is ignored in favor of the overloads everywhere else.

Reply:
Yes, that's correct, but shouldn't the function return be type checked so that future editors don't accidentally make a mistake?

Reply:
I think best we can do is add a comment.